### DISCUSSION TITLE
How to convert images to 512 vectors instead of 2048 vectors

### DISCUSSION THREAD

**Discussion by zhenzi0322** [towhee-io/towhee#2097] [Q&A]
How to convert images to 512 vectors instead of 2048 vectors.

```python
feat = towhee.glob(img_path) \
            .image_decode() \
            .image_embedding.timm(model_name='resnet50') \
            .tensor_normalize() \
            .to_list()
```
At present, the converted DIM is 2048. How to convert it to 512DIM?

**Comment by shiyu22** [comment]
1. Dimensionality Reduction
You can refer to [this notebook](https://github.com/towhee-io/examples/blob/main/image/reverse_image_search/2_deep_dive_image_search.ipynb) to reduce the dimension, and I think you need to save the `projection_matrix` to ensure that you are always trans with one matrix.

2. Try more models 
You can try some other models with timm operator, such as vgg16:
```bash
feat = towhee.glob(img_path) \
            .image_decode() \
            .image_embedding.timm(model_name='vgg16') \
            .tensor_normalize() \
            .to_list()
```
And you can also list all timm model and try with them:
```
import timm
timm.list_models()
```

**Reply by zhenzi0322** [reply]
ok
